### ISSUE TITLE
A user can't select a node ID when deploying without knowing the name of the farm

### ISSUE THREAD

**Issue by LeeSmet** [threefoldtech/tfgrid-sdk-ts#1094]
### Description

Currently, the node ID input when deploying is filled with the nodes belonging to the select farm and a user can't select any other node ID. This means that a user needs to know the name of the farm to which the node belongs if he wants to deploy on a specific node. 

Users who have a specific node ID to deploy on often don't have a farm name however. Some users just use a specific node they already used, and don't care about the farm name. Others might have looked up the node on the dashboard, noted its node ID, but forgotten the name of the farm already since they don't consider that relevant information if you want to deploy on a specific node.

The user should not be constrained to nodes from a specific farm. The farm ID is part of a node object in gridproxy, so if a user enters a node which is not part of the selected farm (if a farm is selected in the first place), looking up the node and then setting the farm to the proper value would improve the UX for these users.

**Comment by RoberL72** [comment]
In other words; the way it works on the current 'old' playground should remain.

**Comment by maayarosama** [comment]
Blocked on the ui design from @ehab-hassan.

**Comment by ehab-hassan** [comment]
Test the Radio buttons at the Node selection
https://www.figma.com/proto/A16wcW2xqLCWWxkj9NduRt/Playground-UI%2FUX?type=design&node-id=262-654&t=Zx5yRtXuT0ibRg47-0&scaling=min-zoom&page-id=0%3A1

I think this is more clear for the user

**Comment by maayarosama** [comment]
UI is done


![Image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/66883096/97881a45-d35c-4e0a-b9f2-77e43d0ffe2a)

WIP:
- the logic behind the manual selection.

**Comment by maayarosama** [comment]
WIP:
- fetching all nodes works fine, however I'm still investigating why reloading nodes doesn't work when switching between way of selection

**Comment by maayarosama** [comment]
Updates:
- The logic behind switching between manual and automated selection
- The Node Selector component is now added to all solutions
- pr is ready and waiting for review

**Comment by maayarosama** [comment]
Done with:
- Changed the node Id field from a select field to a text field
WIP:
- Add a check to see if the node is rented by someone else

**Comment by maayarosama** [comment]
No Updates as I was working on other stuff, will get back to it tomorrow

**Comment by maayarosama** [comment]
Blocked on https://github.com/threefoldtech/tfgrid-sdk-go/issues/420

**Comment by maayarosama** [comment]
As per discussion with @AhmedHanafy725 
- A check should be added in case the selected node is rented by another user.
- A Free flag will be added for checking if the node belongs to a dedicated farm @Omarabdul3ziz. (blocked)
- Manual filtering in case of manual selection will be added for a more specific error message

**Comment by maayarosama** [comment]
Work Completed:
- Added a check in case the selected node is rented by another user.

Work in Progress (WIP):
- Manual filtering in case of manual selection will be added for a more specific error message

**Comment by maayarosama** [comment]
Work Completed:
- Added more specific error messages in case of manual filtering

Work in Progress (WIP):
 - Adding a check for InDedicatedFarm to check if the node belongs to a dedicated farm or not. blocked on deployment of https://github.com/threefoldtech/tfgrid-sdk-go/issues/483

**Comment by maayarosama** [comment]
Work Completed:

- Fixed build errors.
- used Selection enum across all solutions.
- enhancement to the UI.

Work in Progress (WIP):
- Dividing the validations into smaller methods. 

Investigation and Solution:
- why there're no nodes being retrieved on mount although there's a returned farm.

**Comment by maayarosama** [comment]
Work Completed:

Work in Progress (WIP):
- Dividing the validations into smaller methods.

Investigation and Solution:
- Figuring out that the reason behind no nodes being retrieved on mount  is the the added comparison between the old and new values of diskSize arrays
 ```
value.diskSizes.length === oldValue.diskSizes.length &&
new Set([...value.diskSizes, ...oldValue.diskSizes]).size === value.diskSizes.length &&
```
if these comparisons were to be removed the  the deploy button enable/disable bug in the manual selection will be back.
Another solution for this is to figure out why farmManger doesn't subscribe on first try (still investigating)

**Comment by maayarosama** [comment]
1. Work Completed:
- Added a check for node status
- Added a check for valid integers
2. Work in Progress (WIP):

3. Investigation and Solution:
- Investigating he temporary appearance of the NaN keyword

**Comment by Mik-TF** [comment]
Based on the new merged PRs and the playground dev, I think this can be closed. Everything is there on the Playground.

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/77026219/eddcb71a-9b71-4259-842d-9b0ee1bb380f)

**Comment by A-Harby** [comment]
Verified,
Devnet, 2.3.0-alpha3.

[TC2445 - TC2444 - Filter Nodes Manually](https://app.testlodge.com/a/26076/projects/40893/suites/234374?expand_section=422336#case_4164692)

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/5c2c3059-8548-46c0-9786-9be1e4e35063)
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/16f45633-ae80-4bfd-b8a7-297f7fc5cfc1)
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/385f71e1-a260-41a9-984b-528e6458d066)

**Comment by ramezsaeed** [comment]
@A-Harby 
more scenarios need to be tested and covered here, check all the comments
Ex.: https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1094#issuecomment-1810302943

**Comment by A-Harby** [comment]
Rented by someone else.

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/35c7f1d5-1722-47db-aa0b-b00a4d6e7cb9)

[TC2445 - TC2444 - Filter Nodes Manually](https://app.testlodge.com/a/26076/projects/40893/suites/234374?expand_section=422336#case_4164692)

**Comment by ramezsaeed** [comment]
Verified:
- Devnet
- 2.3.0-alpha3
